### PR TITLE
Update pulse module documentation examples

### PR DESCRIPTION
**Pull request by dwierichs**
**Description of the Change:**
- Minor modifications of the pulse module docstrings, with a focus on the code examples.
- Added `windows` validation and parsing single-tuple `windows` inputs to `rect`.

### CODE REVIEW

File: pennylane/pulse/__init__.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def circuit(params):
         qml.evolve(H)(params, t=[0, 10])
         return qml.expval(qml.PauliZ(0))
 
->>> params = [1.2]
+>>> params = jnp.array([1.2])
```
Comment (lillian542) [inline pennylane/pulse/__init__.py:223]: Why?

**Comment by lillian542** [comment]
Looks great, thanks for finding those! 🎉 

I think this is a good change to `rect`, it's frustrating always to have to nest `windows` it just for the occasion that someone might want to define two of them.

I had just one question (see above).

**Review by Qottmann — APPROVED** [review]
Thanks, looks good @dwierichs 🌷

Approving assuming tests are passing once retriggered

File: pennylane/pulse/__init__.py (hunk 1 above)
Comment (lillian542) [reply pennylane/pulse/__init__.py:223]: For context, I don't think we should encourage users to make `params` a `jnp.array` in examples. Doing so for any Hamiltonian where different callables take a different number of parameters will raise an error. The `params` argument passed to a `ParametrizedHamiltonian` should be a list or tuple - items in the list might be arrays, but `params` is not one.

Comment (dwierichs) [reply pennylane/pulse/__init__.py:223]: Ah, this is just to match the provided output. The other way around, I can make the output a list of a 0d array :)

Comment (lillian542) [reply pennylane/pulse/__init__.py:223]: Perfect, updating output sounds better I think!

**Review by lillian542 — APPROVED** [review]
Looks great, thanks! 🎉

**Pull request merged**